### PR TITLE
Refactor 123Movie (BaseProvider)

### PR DESCRIPTION
**Pull request by Teekzilla**
- This depends on [ApolloTVofficial/kamino#123](https://github.com/ApolloTVofficial/kamino/pull/123) being merged. This implementation assumes that the year is passed as a separate parameter and not with the title itself.

### CODE REVIEW

File: src/scrapers/providers/universal/123movie.js [hunk 1]
```diff
@@ -1,133 +1,80 @@
...
-            // targetPageHTML is the HTML for the player page.
-            let targetPageHTML = await rp({
-                uri: pageURL,
-                timeout: 5000,
-                headers: {
-                    'Referer': videoPage
-                }
-            });
-
+            let targetPageHTML = await this._createRequest(rp, pageURL, jar, headers);
```
Comment (ddivad) [inline src/scrapers/providers/universal/123movie.js]: Here, `headers` has the referrer of `url`, which is the root url. However, before it had referrer of `videoPage`.

Does this cause any issues, or does it work ok having `url` as the referrer instead?

File: src/scrapers/providers/universal/123movie.js [hunk 2]
```diff
@@ -1,133 +1,80 @@
...
-                uri: outputPageURL,
-                timeout: 5000,
-                headers: {
-                    'Referer': videoPage
-                }
-            });
-
-            // This is the provider URL
-            //let rapidVideoURL = cheerio.load(outputPageHTML)('link[rel="canonical"]').attr('href');
+            let outputPageHTML = await this._createRequest(rp, outputPageURL, jar, headers);
```
Comment (ddivad) [inline src/scrapers/providers/universal/123movie.js]: Same here

File: src/scrapers/providers/universal/123movie.js (hunk 1 above)
Comment (Teekzilla) [reply src/scrapers/providers/universal/123movie.js]: It works the same, I don't think it matters. I was trying to minimize changes to the headers. From my testing at least still got the same results.

File: src/scrapers/providers/universal/123movie.js (hunk 2 above)
Comment (Teekzilla) [reply src/scrapers/providers/universal/123movie.js]: Same situation as mentioned above, at least from my testing. I received the same successful results for both movies and TV shows. I'll test again tomorrow to confirm a second time, but before pushing the changes I tested a couple movies and TV shows.

**Pull request merged**